k-distribution/src/main/scripts/bin/krun

```diff
@@ -540,20 +540,33 @@ if [ "$backendName" = "llvm" ]; then
   fi
 
   keepTempsIfDryRun "$tempDir" "$expanded_input_file" "$kore_output"
-  if $search; then
+  if $search || [ -n "$pattern" ]; then
     if [ ! -f "$kompiledDir/search" ] || [ ! -x "$kompiledDir/search" ]; then
       error 'You must pass --enable-search to kompile to be able to use krun --search with the LLVM backend'
     fi
-    if [ -n "$pattern" ]; then
-      error "Backend llvm does not support search patterns yet."
+    patternFile="$(mktemp ${tempDir}/tmp.pattern.XXXXXXXXXX)"
+    tempFiles+=("$patternFile")
+    if [ -z "$pattern" ]; then
+      echo '\and{SortGeneratedTopCell{}}(VarResult:SortGeneratedTopCell{},\top{SortGeneratedTopCell{}}())' > "$patternFile"
+    else
+      k-compile-search-pattern --directory "$dir" "$pattern" > "$patternFile"
     fi
+
     if [[ $searchType != "FINAL" ]]; then
       error "Backend llvm does not support --search-all, --search-one-or-more-steps, or --search-one-step yet"
     fi
+    disjunctFile="$(mktemp ${tempDir}/tmp.disjunction.XXXXXXXXXX)"
+    tempFiles+=("$disjunctFile")
+    if $search; then
+      llvmCommand="$kompiledDir/search"
+    else
+      llvmCommand="$kompiledDir/interpreter"
+    fi
```

Comment:
I don't quite understand this.....

We already said `$search || [ -n $pattern ]]` above. So does this mean you can do a normal `krun --pattern SOME_PATTERN` _without_ it doing a search, and have it try to match the single final configuration with the given pattern and print out the substitution?

Reply:
Yes, if you pass `--pattern` but not `--search`, it will do single-path concrete execution and then try to match the result of such execution against the pattern. This is the same behavior krun has had in the past.

Reply:
Cool, thanks :)